backend-api/src/functions/services/sessionService/sessionService.ts

```diff
@@ -0,0 +1,58 @@
+import { randomUUID } from "crypto";
+import { Result } from "../../utils/result";
+import { DynamoDbAdapter } from "../../adapters/dynamoDbAdapter";
+
+export class SessionService implements ISessionService {
+  readonly dynamoDbAdapter: DynamoDbAdapter;
+
+  constructor(tableName: string) {
+    this.dynamoDbAdapter = new DynamoDbAdapter(tableName);
+  }
+
+  async createSession(
+    attributes: CreateSessionAttributes,
+  ): Promise<Result<string>> {
+    const sessionId = randomUUID();
+    return await this.dynamoDbAdapter.createSession(attributes, sessionId);
+  }
+
+  async getActiveSessionId(
+    subjectIdentifier: string,
+  ): Promise<Result<string | null>> {
+    return await this.dynamoDbAdapter.readSessionId(subjectIdentifier);
+  }
+
+  async getActiveSessionDetails(
+    subjectIdentifier: string,
+  ): Promise<Result<SessionDetails | null>> {
+    return await this.dynamoDbAdapter.readSessionDetails(subjectIdentifier);
+  }
```

Comment:
This is looking so clean Ana!

Reply:
Robert Martin would be proud

Reply:
It's going to be challenging keeping this as clean once we have the the try catch blocks when we throw the error!

maybe 

```
try {
const session = await.this.dynamoDbAdapter.getSessionBySubjectIdentifier(subjectIdentifier)
// validate session response if needs be
catch(error) {
  return generateResultFromError (error)
}


```

Reply:
Can we come back to this if necessary?